pkg/reconciler/containersource/containersource.go

```diff
@@ -244,23 +220,75 @@ func sinkArg(source *v1alpha1.ContainerSource) (string, bool) {
 	return "", false
 }
 
-func (r *Reconciler) getDeployment(ctx context.Context, source *v1alpha1.ContainerSource) (*appsv1.Deployment, error) {
-	dl, err := r.KubeClientSet.AppsV1().Deployments(source.Namespace).List(metav1.ListOptions{})
-	if err != nil {
-		r.Logger.Errorf("Unable to list deployments: %v", zap.Error(err))
-		return nil, err
+func (r *Reconciler) reconcileReceiveAdapter(ctx context.Context, src *v1alpha1.ContainerSource, args resources.ContainerArguments) (*appsv1.Deployment, error) {
+	expected := resources.MakeDeployment(args)
+
+	ra, err := r.KubeClientSet.AppsV1().Deployments(src.Namespace).Get(expected.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		ra, err = r.KubeClientSet.AppsV1().Deployments(src.Namespace).Create(expected)
+		if err != nil {
+			r.markNotDeployedRecordEvent(src, corev1.EventTypeWarning, "DeploymentCreateFailed", "Could not create deployment: %v", err)
+			return nil, fmt.Errorf("creating new deployment: %v", err)
+		}
+		r.markDeployingAndRecordEvent(src, corev1.EventTypeNormal, "DeploymentCreated", "Created deployment %q", ra.Name)
+		return ra, nil
+	} else if err != nil {
+		r.markDeployingAndRecordEvent(src, corev1.EventTypeWarning, "DeploymentGetFailed", "Error getting deployment: %v", err)
+
+		return nil, fmt.Errorf("getting deployment: %v", err)
+	} else if !metav1.IsControlledBy(ra, src) {
+		r.markDeployingAndRecordEvent(src, corev1.EventTypeWarning, "DeploymentNotOwned", "Deployment %q is not owned by this ContainerSource", ra.Name)
+		return nil, fmt.Errorf("deploymtn %q is not owned by ContainerSource %q", ra.Name, src.Name)
+	} else if r.podSpecChanged(ra.Spec.Template.Spec, expected.Spec.Template.Spec) {
+		ra.Spec.Template.Spec = expected.Spec.Template.Spec
+		ra, err = r.KubeClientSet.AppsV1().Deployments(src.Namespace).Update(ra)
+		if err != nil {
+			return ra, fmt.Errorf("updating deployment: %v", err)
+		}
+		return ra, nil
+	} else {
```

Comment:
I think you can get rid of this last else statement and just log

Reply:
I don't think so. For example, if the receive adapter isn't found, then it is created. If we moved the log line outside the else, we would log that we are reusing an existing receive adapter.